L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -142,509 +143,237 @@ L1TrackJetProducer::~L1TrackJetProducer() {}
-    // Note: left neighbor will definitely be used because if it
-    //       didn't belong to me it would have been used already
-    clusters[nclust] = phislice[etabin];
-    phislice[etabin].used = true;
-    if (left_pt > 0) {
-      if (clusters != nullptr) {
-        clusters[nclust].pTtot += left_pt;
-        clusters[nclust].numtracks += phislice[etabin - 1].numtracks;
-        clusters[nclust].numtdtrks += phislice[etabin - 1].numtdtrks;
-        for (unsigned int itrk = 0; itrk < phislice[etabin - 1].trackidx.size(); itrk++)
-          clusters[nclust].trackidx.push_back(phislice[etabin - 1].trackidx[itrk]);
-      }
-    }
-    if (my_pt >= right2pt && right_pt > 0) {
-      if (clusters != nullptr) {
-        clusters[nclust].pTtot += right_pt;
-        clusters[nclust].numtracks += phislice[etabin + 1].numtracks;
-        clusters[nclust].numtdtrks += phislice[etabin + 1].numtdtrks;
-        for (unsigned int itrk = 0; itrk < phislice[etabin + 1].trackidx.size(); itrk++)
-          clusters[nclust].trackidx.push_back(phislice[etabin + 1].trackidx[itrk]);
-        phislice[etabin + 1].used = true;
-      }
-    }
-    nclust++;
-  }  // for each etabin
-
-  // Now merge clusters, if necessary
-  for (int m = 0; m < nclust - 1; ++m) {
-    if (std::abs(clusters[m + 1].eta - clusters[m].eta) < 1.5 * etaStep_) {
-      if (clusters[m + 1].pTtot > clusters[m].pTtot) {
-        clusters[m].eta = clusters[m + 1].eta;
-      }
-      clusters[m].pTtot += clusters[m + 1].pTtot;
-      clusters[m].numtracks += clusters[m + 1].numtracks;  // Previous version didn't add tracks when merging
-      clusters[m].numtdtrks += clusters[m + 1].numtdtrks;
-      for (unsigned int itrk = 0; itrk < clusters[m + 1].trackidx.size(); itrk++)
-        clusters[m].trackidx.push_back(clusters[m + 1].trackidx[itrk]);
-
-      for (int m1 = m + 1; m1 < nclust - 1; ++m1)
-        clusters[m1] = clusters[m1 + 1];
-      nclust--;
-      m = -1;
-    }  // end if clusters neighbor in eta
-  }    // end for (m) loop
-
-  for (int i = nclust; i < etaBins_ / 2; ++i)  // zero out remaining unused clusters
-    clusters[i].pTtot = 0;
-  return clusters;
-}
 
 void L1TrackJetProducer::beginStream(StreamID) {}
 
 void L1TrackJetProducer::endStream() {}
 
-bool L1TrackJetProducer::trackQualityCuts(int trk_nstub, float trk_chi2, float trk_bendchi2) {
+bool L1TrackJetProducer::trackQualityCuts(
+    int trk_nstub, float trk_chi2, float trk_bendchi2) {
   bool PassQuality = false;
-  if (trk_bendchi2 < trkBendChi2Max_ && trk_chi2 < trkChi2dofMax_ && trk_nstub >= 4 && !displaced_)
-    PassQuality = true;
-  if (displaced_ && trk_bendchi2 < nStubs4DisplacedBend_ && trk_chi2 < nStubs4DisplacedChi2_ && trk_nstub == 4)
-    PassQuality = true;
-  if (displaced_ && trk_bendchi2 < nStubs5DisplacedBend_ && trk_chi2 < nStubs5DisplacedChi2_ && trk_nstub > 4)
-    PassQuality = true;
+  if (!displaced_){
+     if ( trk_nstub == 4 && trk_bendchi2 < nStubs4PromptBend_ &&
```

Comment:
Define 4 and use it in the lines below too

Reply:
I added the explanation: tracks with 4 stub are of poor resolution/quality therefore we put usually different cuts than those of high quality. This is why the variables have "Nstubs4" in the name